src/workspaces/common/state/useCloningWorkspaceNotifications.test.tsx

```diff
@@ -0,0 +1,143 @@
+  useCloningWorkspaceNotifications,
+} from 'src/workspaces/common/state/useCloningWorkspaceNotifications';
+import { WorkspaceInfo, WorkspaceState } from 'src/workspaces/utils';
+
+type AjaxContract = ReturnType<typeof Ajax>;
+type AjaxWorkspacesContract = AjaxContract['Workspaces'];
+
+jest.mock('src/libs/ajax', (): typeof import('src/libs/ajax') => {
+  return {
+    ...jest.requireActual('src/libs/ajax'),
+    Ajax: jest.fn(),
+  };
+});
+
+type NotificationExports = typeof import('src/libs/notifications');
+jest.mock<NotificationExports>(
+  'src/libs/notifications',
+  (): NotificationExports => ({
+    ...jest.requireActual('src/libs/notifications'),
+    notify: jest.fn(),
+    clearNotification: jest.fn(),
+  })
+);
+// notify
+const CloningTestComponent = (): React.ReactNode => {
+  useCloningWorkspaceNotifications();
+  return null;
+};
+
+describe('useCloningWorkspaceNotifications', () => {
+  beforeEach(() => {
+    cloningWorkspacesStore.set([]);
+  });
+
+  describe('notifyNewWorkspaceClone', () => {
+    it('adds the workspace to the cloning store', () => {
+      // Arrange
+      const clone: WorkspaceInfo = {
+        ...defaultAzureWorkspace.workspace,
+        state: 'Cloning',
+      };
+      // Act
+      notifyNewWorkspaceClone(clone);
+
+      // Assert
+      expect(cloningWorkspacesStore.get()).toEqual([clone]);
+    });
+
+    it('creates a notification that the clone has started', () => {
+      // Arrange
+      const clone: WorkspaceInfo = {
+        ...defaultAzureWorkspace.workspace,
+        state: 'Cloning',
+      };
+      // Act
+      notifyNewWorkspaceClone(clone);
+
+      // Assert
+      expect(asMockedFn(notify)).toHaveBeenCalledWith('info', expect.any(Object), {
+        id: expect.any(String),
+        message: expect.any(Object),
+      });
+    });
+  });
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  it.each<{
+    updatedState: WorkspaceState;
+    notificationType?: NotificationType;
+  }>([
+    { updatedState: 'CloningFailed', notificationType: 'error' },
+    { updatedState: 'Ready', notificationType: 'success' },
+  ])(
+    'removes the workspace in the store and sends a $notificationType notification when the workspace is updated to $updatedState',
+    async ({ updatedState, notificationType }) => {
+      // Arrange
+      const clone: WorkspaceInfo = { ...defaultAzureWorkspace.workspace, state: 'Cloning' };
+      cloningWorkspacesStore.set([clone]);
+      const mockDetailsFn = jest.fn().mockResolvedValue({ workspace: { state: updatedState } });
+      const mockAjax: DeepPartial<AjaxContract> = {
+        Workspaces: {
+          workspace: () =>
+            ({
+              details: mockDetailsFn,
+            } as Partial<AjaxWorkspacesContract['workspace']>),
+        },
+      };
+      asMockedFn(Ajax).mockImplementation(() => mockAjax as AjaxContract);
+
+      // Act
+      render(<CloningTestComponent />);
+      jest.advanceTimersByTime(30000);
```

Comment:
I think it would be good to have a constant for the refresh rate, so that the test updates if it changes.

And I think we could poll more often than every 30 seconds to avoid the strange behavior I saw where the notification updated long after the workspace actually stopped cloning.

Reply:
The way I fixed the update behavior, was just changing the order the hook are called, so the cloning specific one happens first.
But it probably does make sense to have this one on a faster timer anyway.